pkg/diagnostics/analyzers.go

```diff
@@ -150,6 +156,62 @@ func (a *analyzerFactory) eksaDockerAnalyzers() []*Analyze {
 	return append(analyazers, a.generateDeploymentAnalyzers(deployments)...)
 }
 
+// EksaLogTextAnalyzers given a slice of Collectors will check which namespaced log collectors are present
+// and return the log analyzers associated with the namespace in the namespaceLogTextAnalyzersMap
+func (a *analyzerFactory) EksaLogTextAnalyzers(collectors []*Collect) []*Analyze {
+	var analyzers []*Analyze
+	analyzersMap := a.namespaceLogTextAnalyzersMap()
+	for _, collector := range collectors {
+		if collector.Logs != nil {
+			analyzer, ok := analyzersMap[collector.Logs.Namespace]
+			if ok {
+				analyzers = append(analyzers, analyzer...)
+			}
+		}
+	}
+	return analyzers
+}
+
+// namespaceLogTextAnalyzersMap is used to associated log text analyzers with the logs collected from a specific namespace.
+// the key of the analyzers map is the namespace name, and the value are the associated log text analyzers.
+func (a *analyzerFactory) namespaceLogTextAnalyzersMap() map[string][]*Analyze {
+	logTextAnalyzers := map[string][]*Analyze{}
+	logTextAnalyzers[constants.CapiKubeadmControlPlaneSystemNamespace] = a.capiKubeadmControlPlaneSystemLogAnalyzers()
+	return logTextAnalyzers
+}
+
+func (a *analyzerFactory) capiKubeadmControlPlaneSystemLogAnalyzers() []*Analyze {
+	capiCpManagerPod := "capi-kubeadm-control-plane-controller-manager-*"
+	capiCpManagerContainerLogFile := path.Join(capiCpManagerPod, "manager.log")
```

Comment:
Why not just define this directly in a string literal?

Reply:
do you mean like `fmt.Sprintf("%/manager.log", capiCpManagerPod)`? If so, no real reason, just that it is in fact a path and that seemed sane

if you mean why not just have `log = "capi-kubeadm-control-plane-controller-manager-*/manager.log"`, I wanted to set up a pattern for accessing the logs of containers in pods with multiple containers; in this case, we're not gathering the logs for the rbacproxy container so it does not really matter, but if we wanted to add it it the pod name is already seperated